src/jax.jl

```diff
@@ -26,17 +24,17 @@ end
 function (wrap::JaxFunctionWrapper)(args...)
     # TODO: handle multiple outputs
     out = (wrap.jaxfn(mapover(x->jax.numpy.asarray(PyReverseDims(x)), x-> x isa Array, args)...))
-    return ReverseDimsArray(via_dlpack(dlpack, out))
+    return DLArray(out, pyto_dlpack).data
 end
 
 function ChainRulesCore.rrule(wrap::JaxFunctionWrapper, args...)
     project = ProjectTo(args)
     jax_primal, jax_vjpfun = jax.vjp(wrap.jaxfn, mapover(x->jax.numpy.asarray(PyReverseDims(x)), x-> x isa Array, args)...)
     function JaxFunctionWrapper_pullback(Δ)
-        tangent_vals = mapover(x->ReverseDimsArray(via_dlpack(dlpack, x)), x-> x isa PyObject, jax_vjpfun(jax.numpy.asarray(PyReverseDims(Δ))))
+        tangent_vals = mapover(x->(DLArray(x, pyto_dlpack).data), x-> x isa PyObject, jax_vjpfun(jax.numpy.asarray(PyReverseDims(Δ))))
         return (NoTangent(), project(tangent_vals)...)
     end
-    return ReverseDimsArray(via_dlpack(dlpack, jax_primal)), JaxFunctionWrapper_pullback
+    return (DLArray(jax_primal, pyto_dlpack).data), JaxFunctionWrapper_pullback
```

Comment:
Same as above, `jax_primal` can be garbage collected by python and julia has no control over this.

Reply:
I see. I'll have think a bit about how best to do this then, since we don't want the julia user to think too much about conversions between function calls.

Reply:
Would it be better to implement the "full" `ArrayInterface` for `DLArray` (or I can define another container that wraps the `DLArray`).

Reply:
I was hesitant on implementing the whole `ArrayInterface` for `DLArray`, mainly beacuse it will have to cover arbitrary things like broadcast and what not, that the user might want to do with it (possibly in combination with other type of arrays). Let me think about it.

Reply:
Maybe I can simply forward everything to the `data` field, and it will work well enough in general. I'll write a PR.

Reply:
https://github.com/FluxML/MacroTools.jl/blob/master/src/examples/forward.jl should work?

Reply:
Yes, that can work, but I think the number of methods I need to implement is actually small, and probably I shouldn't have worried about it.